Problem: sdk 0.50 is not used

yihuang opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

mmsqe reviewed (commented):
do we need regen protos with cosmossdk.io for customtype?

yihuang commented:
> do we need regen protos with cosmossdk.io for customtype?

yes, still downloading docker images.

mmsqe commented on go.mod:
```diff
@@ -1,223 +1,246 @@
 )
 
 replace (
+	cosmossdk.io/collections => github.com/mmsqe/cosmos-sdk/collections v0.0.0-20240307052158-b7a6ec132cd3
```
can wait [this](https://github.com/cosmos/cosmos-sdk/pull/19675) fix a bit

mmsqe commented on go.mod:
```diff
@@ -1,223 +1,250 @@
-	github.com/cometbft/cometbft-db => github.com/crypto-org-chain/cometbft-db v0.0.0-20231011055109-57922ac52a63
-	github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.46.0-beta2.0.20231013072415-eec017435467
+	github.com/cockroachdb/pebble => github.com/cockroachdb/pebble v0.0.0-20230209160836-829675f94811
+	github.com/cosmos/cosmos-sdk => github.com/mmsqe/cosmos-sdk v0.46.0-beta2.0.20240315010123-8531387f70f0
```
test eth priority pass, can wait the [fix](https://github.com/cosmos/cosmos-sdk/pull/19759) a bit

yihuang commented on x/evm/keeper/keeper.go:
```diff
@@ -18,13 +18,14 @@ package keeper
 	errorsmod "cosmossdk.io/errors"
-	"github.com/cometbft/cometbft/libs/log"
+	"cosmossdk.io/log"
+	sdkmath "cosmossdk.io/math"
```
```suggestion
```

Commit:
Problem: sdk 0.50 is not used

fix dependencies

fix build

upgrade protogen image

Co-authored-by: mmsqe <mavis@crypto.com>

wait for block properly

fix py-lint

fix test

fix py-lint
